Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+
+                        //if (!isNotPosAnchorEnd)
+                        //{
+                        //    v.SetRelativeLayout(dimension == Dimension.Width ? (new Size (autoMin, 0)) : new Size (0, autoMin));
+                        //}
+
+                        if (isNotPosAnchorEnd)
+                        {
+                            int size = dimension == Dimension.Width ? v.Frame.X + v.Frame.Width : v.Frame.Y + v.Frame.Height;
+
+                            if (size > subviewsSize)
+                            {
+                                subviewsSize = size;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // All sizes here are content-relative; ignoring adornments.
+        // We take the larger of text and content.
+        int max = int.Max (textSize, subviewsSize);
+
+        // And, if min: is set, it wins if larger
+        max = int.Max (max, autoMin);
+
+        // Factor in adornments
+        Thickness thickness = us.GetAdornmentsThickness ();
+
+        if (dimension == Dimension.Width)
+        {
+            max += thickness.Horizontal;
+        }
+        else
+        {
+            max += thickness.Vertical;
+        }
+
+        // If max: is set, clamp the return - BUGBUG: Not tested
+        return int.Min (max, MaximumContentDim?.Anchor (superviewContentSize) ?? max);
+    }
+
+    internal override bool ReferencesOtherViews ()
+    {
+        // BUGBUG: This is not correct. _contentSize may be null.
+        return false; //_style.HasFlag (DimAutoStyle.Content);
+    }
+}
+
+/// <summary>
+///     Represents a dimension that is a combination of two other dimensions.
+/// </summary>
+/// <param name="add">
+///     Indicates whether the two dimensions are added or subtracted. If <see langword="true"/>, the dimensions are added,
+///     otherwise they are subtracted.
+/// </param>
+/// <remarks>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+/// </remarks>
+/// <param name="left">The left dimension.</param>
+/// <param name="right">The right dimension.</param>
+public class DimCombine (bool add, Dim left, Dim right) : Dim
+{
+    /// <summary>
+    ///     Gets whether the two dimensions are added or subtracted.
+    /// </summary>
+    public bool Add { get; } = add;
+
+    /// <summary>
+    ///     Gets the left dimension.
+    /// </summary>
+    public Dim Left { get; } = left;
+
+    /// <summary>
+    ///     Gets the right dimension.
+    /// </summary>
+    public Dim Right { get; } = right;
+
+    /// <inheritdoc/>
+    public override string ToString () { return $"Combine({Left}{(Add ? '+' : '-')}{Right})"; }
+
+    internal override int Anchor (int size)
+    {
+        int la = Left.Anchor (size);
```

Comment:
These are costly because at least 3 copies happen, by the end of the method, whether they get used or not.

Should be in-lined at minimum to remove 1-2 copies.

I see why you have the Anchor as a method, but it should be renamed if it stays that way.